### ISSUE TITLE
Documentation on running a simple test on `kaitai_struct_awkward_runtime`

### ISSUE THREAD

**Issue by zonca** [det-lab/kaitai_struct_awkward_runtime#13]
@ManasviGoyal please let me know when the documentation at:
https://github.com/ManasviGoyal/kaitai_struct_awkward_runtime#readme
is ready for me to try running it.
Also please recommend a simple test I can do to make sure everything is working correctly.

**Comment by ManasviGoyal** [comment]
Hi @zonca , I have updated the steps on the `README.md`. Please follow them to run a test for `animal.ksy`. Let me know if you face any issues in following these steps. If this works, there are several other cases - `numpy.ksy`, `lists.ksy`, `records.ksy`, `fake.ksy`, and `pixie4e.ksy` that work. So, I think you can set up automated tests using these examples. I have several other test `.ksy` files but don't have the data for them.

Thanks!

**Comment by ManasviGoyal** [comment]
@zonca Also, we discussed in the meeting regarding some things you can work on. 

I tried to merge `kaitai_struct_compiler` upstream and it’s failing. Maybe you can have a look what's causing this. With the upstream, we also need to reflect the new changes in `CppCompiler.scala` to `AwkwardCompiler.scala` since the later is a modified copy of the former.

Another, good to have feature is using `IndexedBuilder` for `EnumType` cases. For now, they are treated as a `NumpyBuilder`.

Feel free to message me here or on slack if you have any questions. Thanks.

**Comment by zonca** [comment]
> @zonca Also, we discussed in the meeting regarding some things you can work on.
> 
> I tried to merge `kaitai_struct_compiler` upstream and it’s failing. Maybe you can have a look what's causing this. With the upstream, we also need to reflect the new changes in `CppCompiler.scala` to `AwkwardCompiler.scala` since the later is a modified copy of the former.

can you please open a separate issue for this?
 
> Another, good to have feature is using `IndexedBuilder` for `EnumType` cases. For now, they are treated as a `NumpyBuilder`.

and another one for this, so we keep things separate

**Comment by ManasviGoyal** [comment]
> can you please open a separate issue for this?

Sure I will do that! Do let me know once you have tried running the steps and in case you get any errors. Thanks.

**Comment by zonca** [comment]
in step 2 why do you checkout the `ManasviGoyal/test` branch? shouldn't we run tests on the `main` branch?

**Comment by zonca** [comment]
@ManasviGoyal ok, I was able to run all the steps in the README,

I have a quick permissions fix in #18 

Then this issue:

> in step 2 why do you checkout the `ManasviGoyal/test` branch? shouldn't we run tests on the `main` branch?

What steps do you take to create the extra files in the test branch? can we do that on the fly as part of the testing process?

**Comment by zonca** [comment]
also, the output at the last step has a prefix:

```python
[{'animalA__Zentry': [{'animalA__ZentryA__Zstr_len': 3, 'animalA__ZentryA__Zspecies': 'cat', 
'animalA__ZentryA__Zage': 5, 'animalA__ZentryA__Zweight': 12},
{'animalA__ZentryA__Zstr_len': 3, 'animalA__ZentryA__Zspecies': 'dog',
 'animalA__ZentryA__Zage': 3, 'animalA__ZentryA__Zweight': 43}, 
{'animalA__ZentryA__Zstr_len': 6, 'animalA__ZentryA__Zspecies': 'turtle', 
'animalA__ZentryA__Zage': 10, 'animalA__ZentryA__Zweight': 5}]}]
```

Yes, this is how it should be. I forget to change the updated output.

**Comment by ManasviGoyal** [comment]
> @ManasviGoyal ok, I was able to run all the steps in the README,
> 
> I have a quick permissions fix in #18
> 
> Then this issue:
> 
> > in step 2 why do you checkout the `ManasviGoyal/test` branch? shouldn't we run tests on the `main` branch?
> 
> What steps do you take to create the extra files in the test branch? can we do that on the fly as part of the testing process?

The only reason I have the separate test branch is that the code is not yet merged to the kaitai repository and that is why it is easier to do the test by adding the submodule and the `kaitai-struct compiler`. The aim is to eventually remove these extra files. This is also linked to Issue #15.
